### PR TITLE
[IoT] Support IoT Hub and DPS Data Residency feature

### PR DESCRIPTION
**Pull request by c-ryan-k**
**Description**

This PR updates the control-plane SDKs for both IoT Hub and DPS to support new API-versions, as well as a new creation property to enforce data residency on these resources.

**Testing Guide**

`az iot hub|dps create -g resource_group -n iot_hub --enforce-data-residency`: Creates an IoT Hub with data-residency enforced (and thereby disabling cross-region disaster recovery).

**History Notes**

[IoT] `az iot hub create`: Add the `--enforce-data-residency` parameter to support creating resources with data residency enforced (and cross-region disaster recovery disabled).
[IoT] `az iot dps create`: Add the `--enforce-data-residency` parameter to support creating resources with data residency enforced (and cross-region disaster recovery disabled).

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
IoT

**Comment by c-ryan-k** [comment]
Looks like some external tests are failing since we updated the default control-plane API-version for hub, will update and push those recordings in a bit

**Review by zhoxing-ms — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/iot/_utils.py [hunk 1]
```diff
@@ -57,3 +57,40 @@ def generate_key(byte_length=32):
...
     return base64.b64encode(token_bytes).decode("utf8")
+
+
+def _dps_certificate_response_transform(certificate_response):
+    from azure.mgmt.iothubprovisioningservices.models import (CertificateListDescription,
+                                                              CertificateResponse,
+                                                              VerificationCodeResponse)
+    if isinstance(certificate_response, CertificateListDescription) and certificate_response.value:
+        for cert in certificate_response.value:
+            cert = _replace_certificate_bytes(cert)
```
Comment (digimaun) [inline src/azure-cli/azure/cli/command_modules/iot/_utils.py:68]: is this assignment needed?

Comment (digimaun) [inline src/azure-cli/azure/cli/command_modules/iot/_utils.py:68]: nit, is this assignment needed?

**Review by wangzelin007 — APPROVED** [review]

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**